tests/test_dataset.py

```diff
@@ -643,3 +643,47 @@ def test_to_gemmi_withNans(data_merged):
     roundtrip = rs.DataSet.from_gemmi(data_merged.to_gemmi())
 
     assert pd.isna(roundtrip.loc[data_merged.index[0], "N(+)"])
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    rs.summarize_mtz_dtypes(False)[["MTZ Code", "Name", "Class"]].melt().itertuples(),
+)
+def test_select_mtzdtype(data_merged, dtype):
```

Comment:
The only thing I don't like about this test is that it doesn't hit the non-anomalous dtypes as far as I can tell. 
May I suggest, 
```python
@pytest.mark.parametrize(
    "dtype",
    rs.summarize_mtz_dtypes(False)[["MTZ Code", "Name", "Class"]].melt().itertuples(),
)
@pytest.mark.parametrize("anomalous", [True, False])
def test_select_mtzdtype(data_merged, dtype, anomalous):
    if not anomalous:
        data_merged = data_merged.stack_anomalous()
```
?

Reply:
`data_merged` includes IMEAN / SIGIMEAN columns, so non-anomalous dtypes are hit by this test:
```
mtz.dtypes:

FreeR_flag              MTZInt
IMEAN                Intensity
SIGIMEAN                Stddev
I(+)          FriedelIntensity
SIGI(+)         StddevFriedelI
I(-)          FriedelIntensity
SIGI(-)         StddevFriedelI
N(+)                    MTZInt
N(-)                    MTZInt
```
However, instead of doing the `stack_anomalous()` call, I suggest changing the test to use the `data_hewl` fixture -- this includes both merged/unmerged data, so it covers more dtypes.